File: packages/core/rules/test/shadow-server/positive.js
// Test cases that SHOULD be detected by shadow-server rules

const { Server } = require('@modelcontextprotocol/sdk/server');
const { StdioServerTransport } = require('@modelcontextprotocol/sdk/server/stdio');

// mcp-server-no-auth - should detect
const server = new Server({ name: "test" });
server.connect(transport);

// mcp-server-insecure-transport - should detect
new StdioServerTransport();

// mcp-hardcoded-credentials - should detect
const config = { apiKey: "EXAMPLE_API_KEY_DO_NOT_USE", password: "secret123" };
const API_KEY = "hardcoded-key-value";
const PASSWORD = "hardcoded-password";

// mcp-unrestricted-cors - should detect
res.setHeader("Access-Control-Allow-Origin", "*");
const corsConfig = { cors: { origin: "*" } };

// mcp-disabled-tls-verification - should detect
const httpsOptions = { rejectUnauthorized: false };
process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";

// mcp-debug-mode-production - should detect
const appConfig = { debug: true };
process.env.DEBUG = "*";

File: packages/core/rules/test/shadow-server/positive.py
# Test cases that SHOULD be detected by shadow-server rules

import requests
import ssl
import urllib.request

# mcp-python-hardcoded-secrets - should detect
API_KEY = "EXAMPLE_API_KEY_DO_NOT_USE"
PASSWORD = "secret123"
SECRET = "hardcoded-secret"

# mcp-python-ssl-disabled - should detect
response = requests.get("https://example.com", verify=False)
requests.post("https://api.example.com", data={}, verify=False)
urllib.request.urlopen("https://example.com", context=ssl._create_unverified_context())

# mcp-python-debug-enabled - should detect
app.debug = True
app.run(debug=True)
